Dockerfile

```diff
@@ -176,8 +176,22 @@ COPY vendor /go/src/github.com/docker/docker/vendor
 RUN go install -v github.com/cpuguy83/go-md2man \
 	github.com/BurntSushi/toml/cmd/tomlv
 
+# Build the tool for embedding resources in Windows binaries
+ENV RSRC_COMMIT e48dbf1b7fc464a9e85fcec450dddf80816b76e0
+RUN set -x \
+    && git clone https://github.com/akavel/rsrc.git /go/src/github.com/akavel/rsrc \
+    && (cd /go/src/github.com/akavel/rsrc \
+          && git checkout -q $RSRC_COMMIT \
+          && go build -v -o /go/bin/rsrc)
+
 # Wrap all commands in the "docker-in-docker" script to allow nested containers
 ENTRYPOINT ["hack/dind"]
 
 # Upload docker source
 COPY . /go/src/github.com/docker/docker
+
+# Compile the Windows resources into the sources
+RUN rsrc \
+       -manifest /go/src/github.com/docker/docker/hack/make/resources/docker.exe.manifest \
+       -ico /go/src/github.com/docker/docker/hack/make/resources/docker.ico \
+       -o   /go/src/github.com/docker/docker/pkg/winresources/rsrc.syso
```

Comment:
So handling this tool as an actual dependency of building-for-Windows is going to be a little tricky (although doable).

We'll probably want to move this kind of stuff into the `autogen` script, and add this tool as an explicitly installed tool on our Windows Jenkins workers, unfortunately.  We'll probably also need to update `project/PACKAGERS.md` to note that it's required (either only when building for Windows, or generally, depending on how we do the script changes to actually generate the necessary code).